examples/component-example.py

```diff
@@ -0,0 +1,122 @@
+import pygfx as gfx
+from wgpu.gui.auto import run, WgpuCanvas
+
+from collagraph import Collagraph, Component, create_element as h, EventLoopType
+from collagraph.renderers import PygfxRenderer
+
+
+class Track(Component):
+    track_geometry = gfx.cylinder_geometry()
+    track_material = gfx.MeshPhongMaterial(color=[1, 0.5, 1])
+
+    def render(self):
+        return h(
+            "Mesh",
+            {
+                **self.props,
+                "geometry": Track.track_geometry,
+                "material": Track.track_material,
+            },
+        )
+
+
+class Scrubber(Component):
+    scrubber_geometry = gfx.sphere_geometry()
+    scrubber_material = gfx.MeshPhongMaterial(color=[0.5, 1, 0.5])
+
+    def __init__(self, props):
+        super().__init__(props)
+        self._captured = False
+        self._sphere = None
+        self._mouse_pos = (0, 0)
+        self.position = [0, 0, 0]
```

Comment:
so, if I understand the `__setattr__` and `__getattribute__` overrides in `Component` correctly, `self.position` is state, but the private fields are not? unless you instantiate `Scrubber` with a `position` prop, then it's not state, but a prop. Right?

Reply:
No more `__setattr__` and `__getattribute__` tricks!